WordPress/Classes/ViewRelated/Post/PrepublishingViewController.swift

```diff
@@ -164,6 +224,38 @@ class PrepublishingViewController: UITableViewController {
         tableView.reloadData()
     }
 
+    private static func option(for identifier: PrepublishingIdentifier) -> PrepublishingOption {
```

Comment:
It seems a bit odd that this is static, I think you can change options to be a var, like this:

```
private var options: [PrepublishingOption] = []
```

And then in the constructor:

```
    init(post: Post, identifiers: [PrepublishingIdentifier], completion: @escaping (CompletionResult) -> ()) {
        self.post = post
        self.completion = completion
        super.init(nibName: nil, bundle: nil)
        self.options = identifiers.map { identifier in
            return option(for: identifier)
        }
    }
```

This won't affect the lifecycle and will keep the VC encapsulation.

Reply:
Ah, good point @leandroalonso. I just hate having these properties be mutable when they don't need to be. I often rely on whether the property is mutable or not when debugging.

What do you think about doing a new initializer on `PrepublishingOption` which takes the identifier? That seems like the cleanest solution to me.

```swift
private extension PrepublishingOption {
    init(identifier: PrepublishingIdentifier) {
        switch identifier {
        case .title:
            self.init(id: .title, title: PrepublishingViewController.Constants.titlePlaceholder, type: .textField)
        case .schedule:
            self.init(id: .schedule, title: PrepublishingViewController.Constants.publishDateLabel, type: .value)
        case .categories:
            self.init(id: .categories, title: NSLocalizedString("Categories", comment: "Label for Categories"), type: .value)
        case .visibility:
            self.init(id: .visibility, title: NSLocalizedString("Visibility", comment: "Label for Visibility"), type: .value)
        case .tags:
            self.init(id: .tags, title: NSLocalizedString("Tags", comment: "Label for Tags"), type: .value)
        }
    }
}
```

Reply:
I'm ok with this solution. :)